packages/iov-bns/src/encode.ts

```diff
@@ -142,7 +143,7 @@ function buildAddAddressToUsernameTx(tx: AddAddressToUsernameTx): codecImpl.app.
   };
 }
 
-function buildSendTransaction(tx: SendTransaction): codecImpl.app.ITx {
+function buildSendTransaction(tx: SendTransaction & WithCreator): codecImpl.app.ITx {
```

Comment:
It would be nice if this function whould check that creator and sender match, preferably at byte level

```
const {prefix1, data1} = decodeBnsAddress(identityToAddress(tx.creator))
vs.
const {prefix2, data2} = decodeBnsAddress(tx.sender)
```

Reply:
Do we want this? I thought the point was that the sender might not be the creator.

Reply:
Right, I'd consider this a "not yet supported" error. We could also choose to support it but then we need to fix the implementation (`src` must be set to the sender, not the creator) and write a test.

Reply:
Let's go with "not yet supported" for now and talk to Martin about implementation.

Reply:
Does this also apply to other chains?

Reply:
Yes. I guess on the other chains it is very unrealistic that sender != creator is possible so the error message can be permanent there.